### PR TITLE
Fix tflint - switch from reviewdog/action-tflint to terraform_tflint

### PR DESCRIPTION
**Pull request by niroz89**
Support for monorepo is not excellent in the reviewdog design. We need to set the working directory and when there is change to multiple env, have to rewrite to support matrix build.

https://github.com/reviewdog/action-tflint/issues/25

To Do:

- [ ] check on automatic init when run with --modules flag ( https://github.com/antonbabenko/pre-commit-terraform/issues/87 )
- [ ] add options to skip warning
- [ ] clean code

When --modules disabled,
https://github.com/SPHTech/bt-drupal-infra/actions/runs/3145865576/jobs/5113661015
image

When --modules enabled.
https://github.com/SPHTech/bt-drupal-infra/actions/runs/3145560001/jobs/5113420619
image

### CODE REVIEW

**Comment by wayne-root** [comment]
Should we keep the warnings enabled? I feel addressing them helps keep the code concise. 

Also if we were to switch to this version of tflint, repositories having their checks ok'ed before will be blocked if tflint issues starts popping up. Will cause disruption if there's a huge list of errors to fix. Suggest we have a `on.workflow_dispatch.inputs` to allow manual skips of tflint checks. But set default to not skip. We enable this switch for a set period of time allowing code owners time to fix any issues blocking their checks before we remove the option.

**Comment by lawliet89** [comment]
@niroz89 let's get this in?

**Comment by niroz89** [comment]
> @niroz89 let's get this in?

think it was failing even for tflint warning, and the exit code for it is `2` or `3` .. it's not handled in this code yet..

File: .github/workflows/terraform.yaml
Comment (niroz89) [inline .github/workflows/terraform.yaml]: Let's pin to specific version rather than latest.

Comment (thepoppingone) [reply .github/workflows/terraform.yaml]: pinned

**Comment by wayne-root** [comment]
Im good with the changes, has been tested on awedio-infra repo. @niroz89 take one last pass at the changes?

**Comment by thepoppingone** [comment]
Verified working on the following cases:
No errors - passes - https://github.com/SPHTech/awedio-infra/actions/runs/3936747293
Errors with skipped true - passes https://github.com/SPHTech/bt-drupal-infra/actions/runs/3936944181
Errors with default skipped false - fails https://github.com/SPHTech/bt-drupal-infra/actions/runs/3936993043

**Comment by niroz89** [comment]
I'm good with the changes, I can't approve this PR as the initial PR creator.. @lawliet89 over to you.

**Review by lawliet89 — APPROVED** [review]

**Pull request merged**
